metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
A lively discussion :-)  There seems to be a proposal to limit the scope of this PR to moving the grokPatterns into configuration, and open a discussion on dev as to how to implement automated parser configuration update.  I am +1 on that.  I am (non-binding) -1 on changing the parse() and parseOptional() interfaces to take a ParseConfiguration on every message.  Configuration and Parsing are currently clearly separated in the structure of the code, and I think it should stay that way.  With your agreement, I'll open the dev discussion with that reasoning.